inspire_utils/dedupers.py

```diff
@@ -68,3 +68,30 @@ def _freeze(o):
             seen.add(f)
 
     return result
+
+
+def dedupe_all_lists(obj, exclude_keys=(), dedupe_top_level=True):
+    """Recursively remove duplicates from all lists.
+
+    Args:
+        obj: collection to deduplicate
+        exclude_keys (Container[str]): key names to ignore for deduplication
+        dedupe_top_level (bool): whether the top-level list should be deduplicated too
+    """
+    squared_dedupe_len = 10
```

Comment:
would it make sense to have this configurable?

Reply:
would it? is anyone going to run microbenchmarks to optimize this value for each specific use case? it's just a performance optimization (that I copied over from `inspire-dojson`), it doesn't affect the results.